### PR TITLE
fix(populate): multiple populates when missing child key

### LINKED ISSUE #277: bug(populate): missing childKey causes no population
**Do you want to request a *feature* or report a *bug*?**  

*bug*

**What is the current behavior?**

Trying with 2.0.0 latest version.

When trying to pass a set of child keys to populate if some of the keys does not exist in the firebase route none of the keys are populated.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [codesandbox](https://codesandbox.io/) or similar.**

I have a /users/uid route in firebase which has only "experiences" key. If non existent childkeys are included in the populates list existing keys should still be populated. 

Populate works fine when I try to initiate with 

```js 
const populates = [
  { child: 'experiences', root: '/allExperiencesList' },
]

const fbWrappedComponent = firebaseConnect(
  ({AuthProfile}) => {
    if(AuthProfile){
      return ([{path: `/users/${AuthProfile ? AuthProfile.uid : ''}`, populates}])
    }
  }
)(TopBarContainer)
```
but fails if I add a key which does not exist and try as follows,

```js
const populates = [
  { child: 'experiences', root: '/allExperiencesList' },
  { child: 'education', root: '/allEducationList' },
]

const fbWrappedComponent = firebaseConnect(
  ({AuthProfile}) => {
    if(AuthProfile){
      return ([{path: `/users/${AuthProfile ? AuthProfile.uid : ''}`, populates}])
    }
  }
)(TopBarContainer)
```
Tried enabling logging to "true" and clearly see that populate is not even triggered in the second case.

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**
As far as I know it did work in previous versions.

### PR DESCRIPTION
**Pull request by JeremyPlease**
### Description

This is a minor update to `populateChild` function in `helpers.js` to returns `null` instead of `child`. This works because the returned value is used in `reduce` function that combines values. 

Previously, already written populates values could be overwritten when `child` was returned.
A new failing test was written to reproduce the issue and is now passing. I've also confirmed this fix in a private repo with `npm link`.

### Check List
If not relevant to pull request, check off as complete

- [x] All tests passing
- [x] Docs updated with any changes or examples if applicable
- [x] Added tests to ensure new feature(s) work properly

### Relevant Issues

Fixes #277.

### CODE REVIEW

**Comment by JeremyPlease** [comment]
Hey @prescottprue, it looks like Travis has an unrelated test failing for node 8. I'm unable to reproduce this failure locally using the same node version at Travis (8.7.0). Could you please trigger a rebuild in Travis to see if it resolves itself?

File: tests/unit/helpers.spec.js [hunk 1]
```diff
@@ -214,6 +235,21 @@ describe('Helpers:', () => {
             .property('displayName', 'scott')
         })
 
+        it('populates value even when others are missing', () => {
```
Comment (prescottprue) [inline tests/unit/helpers.spec.js]: Great work adding the unit test along with it!

File: tests/unit/helpers.spec.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const exampleData = {
     projects: {
       CDF: {
         owner: 'ABC',
+        category: 'cat1',
```
Comment (prescottprue) [inline tests/unit/helpers.spec.js]: Forgot this data was in the file directly. Might end up moving this out to its own file.

Great job figuring it out anyway :)

**Comment by JeremyPlease** [comment]
Thanks, @prescottprue!

**Pull request merged**